app/controllers/active_storage/blobs_controller.rb

```diff
@@ -0,0 +1,39 @@
+# NOTE TO READER: this replacement of ActiveStorage's blob controller
+# is in place to enforce authentication when accessing a blob. At the
+# time of writing, the `/rails/active_storage/blobs/XXX` path is not
+# protected which means the Rails app will happily generate a valid
+# link to the underlying service+asset and redirect anyone who asks
+# there.
+#
+# However ActiveStorage will naturally evolve and probably implement a
+# way to do this without redefining the controller itself but in the
+# meantime it means we've started diverging. This should be kept in
+# sight and ultimately removed.
+#
+# PS: a more direct `before_action: authenticate_user!` was causing an
+# error, probably because the filter is meant to be used in the
+# context of a Rails app rather than its internal engines.
+
+# frozen_string_literal: true
+
+# Take a signed permanent reference for a blob and turn it into an expiring service URL for download.
+# Note: These URLs are publicly accessible. If you need to enforce access protection beyond the
+# security-through-obscurity factor of the signed blob references, you'll need to implement your own
+# authenticated redirection controller.
```

Comment:
This paragraph is a bit confusing as it's stated here, since we are implementing our own authenticated redirection controller, maybe make it explicit this is a reference from the original?